File: src/main/java/com/hypherionmc/sdlink/core/database/HiddenPlayers.java
package com.hypherionmc.sdlink.core.database;

import io.jsondb.annotation.Document;
import io.jsondb.annotation.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Setter
@Getter
@AllArgsConstructor(staticName = "of")
@NoArgsConstructor
@Document(collection = "hiddenplayers", schemaVersion = "1.0")
public class HiddenPlayers {

    @Id
    private String identifier;
    private String displayName;
    private String type;

}

File: Common/src/main/kotlin/com/hypherionmc/sdlinkrw/modules/database/HiddenPlayers.kt
package com.hypherionmc.sdlinkrw.modules.database

import com.hypherionmc.sdlink.core.jsondb.annotations.Document
import com.hypherionmc.sdlink.core.jsondb.annotations.Id

@Document(collection = "hiddenplayers")
data class HiddenPlayers(
    @Id var identifier: String,
    var displayName: String,
    var type: String,
) {
    companion object {
        @JvmStatic
        fun of(identifier: String, displayName: String, type: String): HiddenPlayers {
            return HiddenPlayers(identifier, displayName, type)
        }
    }
}
